src/backend/app/Misc/TransactionDataContainer.php

```diff
@@ -43,11 +44,11 @@ public static function initialize(Transaction $transaction): TransactionDataCont
         try {
             $container->device = $deviceService->getBySerialNumber($transaction->message);
             $container->tariff = null;
-            $container->manufacturer = $container->device->device->manufacturer()->first();
+            $container->manufacturer = $container->device->device->manufacturer ?? null;
 
             if ($container->device->device_type === Meter::RELATION_NAME) {
                 $meter = $container->device->device;
-                $container->tariff = $meter->tariff()->first();
+                $container->tariff = $meter->tariff ?? null;
             }
```

Comment:
I think the issue here is, that Larastan doesn't understand that `$container->device->device` is a Meter in this `if` block.

What about we change this block to

```php
            if ($container->device->device instanceof Meter) {
                $meter = $container->device->device;
                $container->manufacturer = $meter->manufacturer()->first();
                $container->tariff = $meter->tariff()->first();
            }
```

instead?

Reply:
Also, probably want to add 

```
$container->meter = $meter;
```

Else the calls in `AssessRate.php` `$transactionData->meter` will probably fail.